### PR TITLE
HAWQ-1651. Fix some bugs in bulid and unittest

### PR DESCRIPTION
**Pull request by oushu1wangziming1**
HAWQ-1651. Fix some bugs in bulid and unittest.

### CODE REVIEW

File: src/backend/cdb/cdbvars.c [hunk 1]
```diff
@@ -169,7 +169,6 @@ bool gp_interconnect_cache_future_packets=true;
 
 int			Gp_udp_bufsize_k; /* UPD recv buf size, in KB */
 
-#ifdef USE_ASSERT_CHECKING
```
Comment (huor) [inline src/backend/cdb/cdbvars.c]: Do not remove this for assert

File: src/backend/utils/mmgr/mcxt.c [hunk 2]
```diff
@@ -828,7 +828,6 @@ MemoryContextStats(MemoryContext context)
  *
  * This is just a debugging utility, so it's not fancy.
  */
-#ifdef MEMORY_CONTEXT_CHECKING
```
Comment (huor) [inline src/backend/utils/mmgr/mcxt.c]: Do not remove this for assert.

**Comment by huor** [comment]
+1

**Comment by radarwave** [comment]
Per my test this commit fixed several existing issues. +1

File: src/include/nodes/memnodes.h [hunk 3]
```diff
@@ -66,9 +66,7 @@ typedef struct MemoryContextMethods
 	void		(*stats) (MemoryContext context, uint64 *nBlocks, uint64 *nChunks, uint64 *currentAvailable, uint64 *allAllocated, uint64 *allFreed, uint64 *maxHeld);
 	void		(*release_accounting)(MemoryContext context);
 	void		(*update_generation)(MemoryContext context);
-#ifdef MEMORY_CONTEXT_CHECKING
```
Comment (stanlyxiang) [inline src/include/nodes/memnodes.h]: why remove these memory context checking ?

**Comment by stanlyxiang** [comment]
+1

File: src/include/nodes/memnodes.h (hunk 3 above)
Comment (oushu1wangziming1) [reply src/include/nodes/memnodes.h]: If "#ifdef MEMORY_CONTEXT_CHECKING" is not removed,
 MemoryContextCheck couldn't be build in CentOS 6.

**Pull request closed without merging**